### PR TITLE
Use symbolicator proguard processing

### PR DESCRIPTION
**Pull request by loewenheim**
This adds Symbolicator bindings for proguard/JVM processing. All of it is gated behind the config settings `"symbolicator.proguard-processing-projects"` and `"symbolicator.proguard-processing-sample-rate"`.

The implementation passes all the tests in `java/test_plugin`.

Currently, proguard symbolication uses the same Celery queue and Symbolicator pool as native. This may need to change before we use this in earnest.

### CODE REVIEW

File: src/sentry/lang/java/processing.py [hunk 1]
```diff
@@ -23,3 +33,185 @@ def deobfuscate_exception_value(data):
         )
 
     return data
+
+
+def _merge_frame_context(new_frame, symbolicated):
```
Comment (anonrig) [inline src/sentry/lang/java/processing.py:38]: Can you add types to each function, and enable strict type checking using pyproject.toml?

File: src/sentry/lang/java/processing.py [hunk 2]
```diff
@@ -23,3 +33,186 @@ def deobfuscate_exception_value(data):
...
+
+def _handles_frame(frame, data):
+    # Skip frames without function or module
+    return "function" in frame and "module" in frame
+
+
+FRAME_FIELDS = ("abs_path", "lineno", "function", "module", "filename", "in_app")
+
+
+def _normalize_frame(raw_frame: Any) -> dict:
```
Comment (anonrig) [inline src/sentry/lang/java/processing.py:74]:
```suggestion
def _normalize_frame(raw_frame: dict[str, Any]) -> dict[str, Any]:
```

File: src/sentry/lang/java/processing.py [hunk 3]
```diff
@@ -23,3 +33,186 @@ def deobfuscate_exception_value(data):
...
+        new_frame["filename"] = symbolicated["filename"]
+    if symbolicated.get("lineno"):
+        new_frame["lineno"] = symbolicated["lineno"]
+    if symbolicated.get("module"):
+        new_frame["module"] = symbolicated["module"]
+    if symbolicated.get("in_app") is not None:
+        new_frame["in_app"] = symbolicated["in_app"]
+
+
+def _handles_frame(frame, data):
```
Comment (anonrig) [inline src/sentry/lang/java/processing.py:66]:
```suggestion
def _handles_frame(frame, data) -> bool:
```

**Comment by loewenheim** [comment]
Rebased on top of https://github.com/getsentry/sentry/pull/67271 so I can run tests.

File: src/sentry/tasks/symbolication.py [hunk 4]
```diff
@@ -163,11 +168,11 @@ def _continue_to_process_event(was_killswitched: bool = False) -> None:
         # After JS processing, we check `get_native_symbolication_function`,
         # because maybe we need to feed it to another round of
         # `symbolicate_event`, but for *native* that time.
-        if not was_killswitched and task_kind.is_js:
+        if not was_killswitched and task_kind.platform == SymbolicatorPlatform.js:
```
Comment (Swatinem) [inline src/sentry/tasks/symbolication.py:171]: I don’t see any code handling the case the above comment mentions.

The problem here is that all the different tasks delegate to the same internal function.
As well as the fact that we might now need to pass it *three* times through the same function, one after the other, as in theory we could have a mixed stack trace involving *all* the languages.

That BTW is not really "in theory" as React Native on Android does indeed involve all three platforms.

The logic that decides which inner symbolication function to use, and to which other symbolication task to forward the event to is really horrible :-(
Ideally, I would love to have some kind of bitfield for this, which has markers for which platforms / symbolication functions this needs to run through, vs which ones it has already run through. Together with a function that uses this info to then forward the event to the appropriate next task.

File: tests/relay_integration/lang/java/test_plugin.py [hunk 5]
```diff
@@ -497,6 +505,12 @@ def test_basic_resolving(self):
             "org.slf4j.helpers.Util$ClassContextSecurityManager " "in getExtraClassContext"
         )
 
+    @requires_symbolicator
+    @pytest.mark.symbolicator
+    def test_basic_resolving_symbolicator(self):
+        with override_options({"symbolicator.proguard-processing-sample-rate": 1.0}):
+            self.test_basic_resolving()
```
Comment (Swatinem) [inline tests/relay_integration/lang/java/test_plugin.py:512]: IMO its okay to duplicate all these tests. Though ideally you can dig up the code we had previously for the JS rollout to properly parameterize the existing tests.

**Review by Swatinem — COMMENTED** [review]
doing this as a symbolication task indeed looks a lot better than just doing it in the stacktrace processor.

I don’t think this properly handles mixed stack traces right now, see my lengthy comment down below.

File: tests/relay_integration/lang/java/test_plugin.py (hunk 5 above)
Comment (loewenheim) [reply tests/relay_integration/lang/java/test_plugin.py:512]: That unfortunately doesn't work—it used a fixture for parametrization and these tests don't support fixtures (something about `unittest` vs `pytest`).

**Review by Swatinem — APPROVED** [review]
we discussed to land this even though it does not work correctly yet. as long as we are not turning the feature flag on, nothing will break, and a followup PR that is self-contained will solve the last missing piece.

File: src/sentry/lang/java/processing.py [hunk 6]
```diff
@@ -23,3 +35,206 @@ def deobfuscate_exception_value(data):
         )
 
     return data
+
+
+def _merge_frame_context(new_frame: dict[str, Any], symbolicated: dict[str, Any]):
```
Comment (anonrig) [inline src/sentry/lang/java/processing.py:40]: There is no comment in any of the codes. Some might be obvious, some might not. Can you add comments to make the code more maintainable?

**Review by Swatinem — APPROVED** [review]

**Pull request merged**